sklearn/base.py

```diff
@@ -156,6 +156,8 @@ class BaseEstimator:
     at the class level in their ``__init__`` as explicit keyword
     arguments (no ``*args`` or ``**kwargs``).
     """
+    # used by _validate_data when `y` is not validated
+    __NO_Y = '__NO_Y'
```

Comment:
If you really want to keep a module level constant as a default placeholder, I would rather use single underscore. It's enough to tell it's private. But I think the code would be clearer without defining a module level constant.